tests/integration/test_issue_2410.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2024 Pex project contributors.
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import
+
+import os.path
+import subprocess
+from textwrap import dedent
+
+from colors import colors
+
+from pex.common import safe_open
+from pex.typing import TYPE_CHECKING
+from testing import run_pex_command
+
+if TYPE_CHECKING:
+    from typing import Any
+
+
+def test_pex_with_editable(tmpdir):
+    # type: (Any) -> None
+
+    project_dir = os.path.join(str(tmpdir), "project")
+    with safe_open(os.path.join(project_dir, "example.py"), "w") as fp:
+        fp.write(
+            dedent(
+                """\
+                import sys
+
+                import colors
+
+
+                def colorize(*messages):
+                    return colors.green(" ".join(messages))
+
+
+                if __name__ == "__main__":
+                    print(colorize(*sys.argv[1:]))
+                    sys.exit(0)
+                """
+            )
+        )
+    with safe_open(os.path.join(project_dir, "setup.py"), "w") as fp:
+        fp.write(
+            dedent(
+                """\
+                from setuptools import setup
+
+
+                setup(
+                    name="example",
+                    version="0.1.0",
+                    py_modules=["example"],
+                )
+                """
+            )
+        )
+
+    requirements = os.path.join(project_dir, "requirements.txt")
+    with safe_open(requirements, "w") as fp:
+        fp.write(
+            dedent(
+                """\
+                ansicolors==1.1.8
+                -e file://{project_dir}
+                """
+            ).format(project_dir=project_dir)
+        )
+
+    pex = os.path.join(str(tmpdir), "pex")
+    run_pex_command(args=["-r", requirements, "-m", "example", "-o", pex]).assert_success()
+    output = (
+        subprocess.check_output(args=[pex, "A", "wet", "duck", "flies", "at", "night!"])
```

Comment:
Without the fix in `pex/requirements.py` [here](https://github.com/pex-tool/pex/pull/2464/files#diff-75dc165e0a270ddf33c9ab3bb1f7db5728a843926ba90aba86db261deb6f76f7R658) this subprocess fails similarly to the OP in #2410:
```
tests/integration/test_issue_2410.py::test_pex_with_editable Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/home/jsirois/.pex/unzipped_pexes/8a97e1ee50ec37bf439dfaac193544f17e281e0b/__main__.py", line 224, in <module>
    result = boot(
        bootstrap_dir='.bootstrap',
    ...<6 lines>...
        inject_python_args=(),
    )
  File "/home/jsirois/.pex/unzipped_pexes/8a97e1ee50ec37bf439dfaac193544f17e281e0b/__main__.py", line 218, in boot
    bootstrap_pex(
    ~~~~~~~~~~~~~^
        entry_point, python_args=python_args, execute=__SHOULD_EXECUTE__, venv_dir=venv_dir
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/home/jsirois/.pex/unzipped_pexes/8a97e1ee50ec37bf439dfaac193544f17e281e0b/.bootstrap/pex/pex_bootstrapper.py", line 643, in bootstrap_pex
    pex.PEX(entry_point).execute()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/home/jsirois/.pex/unzipped_pexes/8a97e1ee50ec37bf439dfaac193544f17e281e0b/.bootstrap/pex/pex.py", line 562, in execute
    sys.exit(self._wrap_coverage(self._wrap_profiling, self._execute))
             ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jsirois/.pex/unzipped_pexes/8a97e1ee50ec37bf439dfaac193544f17e281e0b/.bootstrap/pex/pex.py", line 469, in _wrap_coverage
    return runner(*args)
  File "/home/jsirois/.pex/unzipped_pexes/8a97e1ee50ec37bf439dfaac193544f17e281e0b/.bootstrap/pex/pex.py", line 500, in _wrap_profiling
    return runner(*args)
  File "/home/jsirois/.pex/unzipped_pexes/8a97e1ee50ec37bf439dfaac193544f17e281e0b/.bootstrap/pex/pex.py", line 606, in _execute
    return self.execute_entry(
           ~~~~~~~~~~~~~~~~~~^
        EntryPoint.parse("run = {}".format(self._pex_info.entry_point))
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/home/jsirois/.pex/unzipped_pexes/8a97e1ee50ec37bf439dfaac193544f17e281e0b/.bootstrap/pex/pex.py", line 810, in execute_entry
    return self.execute_module(entry_point.module)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
  File "/home/jsirois/.pex/unzipped_pexes/8a97e1ee50ec37bf439dfaac193544f17e281e0b/.bootstrap/pex/pex.py", line 818, in execute_module
    runpy.run_module(module_name, run_name="__main__", alter_sys=True)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen runpy>", line 222, in run_module
  File "<frozen runpy>", line 142, in _get_module_details
ImportError: No module named example
FAILED
```